datasets/movielens/data.json

```diff
@@ -0,0 +1,244 @@
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "ratings.csv",
+      "encodingFormat": "text/csv"
+    },
+    {
+      "name": "movies-table",
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "movies.csv",
+      "encodingFormat": "text/csv"
+    },
+    {
+      "name": "tags-table",
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "tags.csv",
+      "encodingFormat": "text/csv"
+    },
+    {
+      "name": "links-table",
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "links.csv",
+      "encodingFormat": "text/csv"
+    },
+    {
+      "name": "genome-scores-table",
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "genome-scores.csv",
+      "encodingFormat": "text/csv"
+    },
+    {
+      "name": "genome-tags-table",
+      "@type": "FileObject",
+      "containedIn": "ml-25m-archive",
+      "contentUrl": "genome-tags.csv",
+      "encodingFormat": "text/csv"
+    }
+  ],
+  "recordSet": [
+    {
+      "name": "movies",
+      "@type": "ml:RecordSet",
+      "source": "#{movies-table}",
+      "key": "#{movie_id}",
+      "field": [
+        {
+          "name": "movie_id",
+          "@type": "ml:Field",
+          "dataType": "Integer",
+          "source": "#{movies-table/movieId}"
+        },
+        {
+          "name": "title",
+          "@type": "ml:Field",
+          "description": "The title of the  movie with the release year in parentheses.",
+          "dataType": "Text",
+          "source": "#{movies-table/title}"
+        },
+        {
+          "name": "genre",
+          "@type": "ml:Field",
+          "description": "A sequence of genres to which the rated movie belongs.",
+          "dataType": "Text",
+          "repeated": "true",
+          "source":
+            {
+              "data": "#{movies-table/genres}",
+              "applyTransform": {"separator": "|"}
+            }
+        }
+      ]
+    },
+    {
+      "name": "ratings",
+      "@type": "ml:RecordSet",
+      "source": "#{ratings-table}",
+      "key": ["#{user_id}", "#{movie_id}"],
+      "field": [
+        {
+          "name": "user_id",
+          "@type": "ml:Field",
+          "dataType": "Integer",
+          "source": "#{ratings-table/userId}"
+        },
+        {
+          "name": "movie_id",
+          "@type": "ml:Field",
+          "dataType": "Integer",
+          "source": "#{ratings-table/movieId}",
+          "references": "#{movies}"
+        },
+        {
+          "name": "rating",
+          "@type": "ml:Field",
+          "description": "The score of the rating on a five-star scale.",
+          "dataType": "Float",
+          "source": "#{ratings-table/rating}"
+        },
+        {
+          "name": "timestamp",
+          "@type": "ml:Field",
+          "dataType": "Integer",
```

Comment:
Do we want all our examples to use semantic datatypes from WikiData, as it's used in https://github.com/mlcommons/datasets_format/blob/main/datasets/titanic/metadata.json? I'm not sure if we should, I'm curious what you think.

If we want semantic datatypes in this document, we could add `wd:186885` to this `dataType` to signify it's a timestamp, and probably add info to the other fields as well.

Reply:
I don' t think we need semantic annotations for basic data types.